### PR TITLE
EVG-17562: decommission pods that have no tasks to dispatch

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-17562

### Description 
* Decommission pods that have no tasks to dispatch. This is just small extra safety in case enqueueing the pod termination job fails to ensure that the pod will refuse to dispatch tasks until it's terminated.
* Allow pod status to be updated with optional reason indicating why the status got updated. The reason makes it more traceable in the event logs why the system took particular actions on the pod.

### Testing 
Updated unit tests.

### CODE REVIEW

**Comment by Kimchelly** [comment]
evergreen retry

**Comment by Kimchelly** [comment]
evergreen retry

File: rest/route/pod_agent.go [hunk 1]
```diff
@@ -304,8 +304,17 @@ func (h *podAgentNextTask) Run(ctx context.Context) gimlet.Responder {
...
 }
 
-// enqueuePodTerminationJob will enqueue the job to terminate a pod with a detailed reason for doing so.
-func (h *podAgentNextTask) enqueuePodTerminationJob(ctx context.Context, reason string) error {
+// prepareForPodTermination will mark the pod as preparing to terminate if it is
+// not already attempting to terminate and enqueue the job to terminate it with
+// a detailed reason for doing so.
+func (h *podAgentNextTask) prepareForPodTermination(ctx context.Context, p *pod.Pod, reason string) error {
+	if p.Status != pod.StatusDecommissioned && p.Status != pod.StatusTerminated {
+		// kim: TODO: test
```
Comment (hadjri) [inline rest/route/pod_agent.go:312]: If this is being implemented later can we ticket and link it here?

**Review by hadjri — APPROVED** [review]

File: rest/route/pod_agent.go (hunk 1 above)
Comment (Kimchelly) [reply rest/route/pod_agent.go]: Ah, this is already tested, I just forgot to remove the note to myself to test it 🤦 . Should be good now.

**Pull request merged**